File: UI_Angular/src/app/services/random-finance-record.service.ts
import { Injectable } from '@angular/core';

import { RandomDescriptionService } from './random-description.service';
import { IFinanceRecord } from '../models/financeRecord';

@Injectable({
  providedIn: 'root'
})
export class RandomFinanceRecordService {

  constructor(private randomDescriptionService: RandomDescriptionService) { }

  generateRandomFinanceRecord(): IFinanceRecord {
    const randomAmount = Math.round(Math.random() * 200 - 100);
    const randomDescription = randomAmount > 0
      ? this.randomDescriptionService.getRandomPositiveDescription()
      : this.randomDescriptionService.getRandomNegativeDescription();

    const randomRecord: IFinanceRecord = {
      id: crypto.randomUUID(),
      date: new Date(),
      amount: randomAmount,
      description: randomDescription,
    };

    return randomRecord;
  }
}

File: UI_Blazor/Services/RandomFinanceRecordService.cs
using UI_Blazor.Models;

namespace UI_Blazor.Services
{
    public class RandomFinanceRecordService
    {
        private RandomDescriptionService randomDescriptionService;

        public RandomFinanceRecordService(RandomDescriptionService randomDescriptionService)
        {
            this.randomDescriptionService = randomDescriptionService;
        }

        public FinanceRecordModel GenerateRandomFinanceRecord()
        {
            var randomAmount = new Random().Next(-100, 100);
            var randomDescription = randomAmount > 0 
                ? randomDescriptionService.GetRandomPositiveDescription()
                : randomDescriptionService.GetRandomNegativeDescription();

            var randomRecord = new FinanceRecordModel
            {
                Id = Guid.NewGuid(),
                Date = DateTime.UtcNow,
                Amount = randomAmount,
                Description = randomDescription
            };

            return randomRecord;
        }
    }
}
